scripts/check-public-api.ts

```diff
@@ -253,3 +251,70 @@ function captureGroupsFromGlobalRegex(regex: RegExp, str: string): string[] {
   const groups = Array.from(str.matchAll(regex));
   return groups.map(group => group[1]);
 }
+
+async function checkBarrelRecursive(cwd: string) {
+  readdirSync(cwd, { withFileTypes: true })
+    .filter(dirent => dirent.isDirectory())
+    .forEach(subDir => {
+      if (subDir.name !== '__snapshots__') {
+        checkBarrelRecursive(join(cwd, subDir.name));
+      }
+    });
+  await exportAllInBarrel(
+    cwd,
+    parse(cwd).name === 'src' ? 'internal.ts' : 'index.ts'
+  );
+}
+
+async function exportAllInBarrel(cwd: string, barrelFileName: string) {
+  const barrelFilePath = join(cwd, barrelFileName);
+  if (existsSync(barrelFilePath) && lstatSync(barrelFilePath).isFile()) {
+    const dirContents = new GlobSync('*', {
+      ignore: [
+        '**/*.spec.ts',
+        '__snapshots__',
+        'internal.ts',
+        'index.ts',
+        'cli.ts',
+        'generator-cli.ts',
+        '**/*.md'
+      ],
+      cwd
+    }).found.map(name => basename(name, '.ts'));
+    const exportedFiles = parseBarrelFile(
+      await readFile(barrelFilePath, 'utf8'),
+      regexExportedInternal
+    );
+    if (!compareBarrels(dirContents, exportedFiles, barrelFilePath)) {
+      process.exit(1);
+    }
+  } else {
+    throw Error(`No ${barrelFileName} file found in ${cwd}`);
+  }
+}
+
+function compareBarrels(
+  dirContents: string[],
+  exportedFiles: string[],
+  barrelFilePath: string
+) {
+  let setsAreEqual = true;
+
+  dirContents.forEach(tsFiles => {
```

Comment:
This is correct but I find the forEach loops a bit long to read. Perhaps you can do a:
``let difference = arrA.filter(x => !arrB.includes(x));`` and do a forEach only for the log statement. Also you can use the difference varialbes (for the two differnces) to sett the boolean return.